src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1242,18 +1290,55 @@ impl Cage {
     }
 
     //------------------------------------FSTAT SYSCALL------------------------------------
+    /// ### Description
+    ///
+    /// `fstat_syscall` retrieves file status information for the file specified
+    /// by the file descriptor `fd` and populates the provided `statbuf`
+    /// with this information.
+    ///
+    /// ### Arguments
+    ///
+    /// It accepts two parameters:
+    /// * `fd` - The file descriptor for the file for which we need the status
+    ///   information.
+    /// * `statbuf` - A mutable reference to a `StatData` struct where the file
+    ///   status will be stored.
+    ///
+    /// ### Returns
+    ///
+    /// For a successful call, the return value will be 0. On error, a negative
+    /// errno is returned to indicate the error.
+    ///
+    /// ### Errors
+    ///
+    /// * `EBADF` - The file descriptor `fd` is invalid.
+    /// * `EOPNOTSUPP` - `fstat` is not supported on sockets.
+    ///
+    /// ### Panics
+    ///
+    /// * If the file descriptor passed is invalid (less than zero or greater
+    ///   than MAX FD which is 1024)
+    /// * If the inode number retrieved from the file descriptor does not exist
+    ///   in `FS_METADATA.inodetable`.
+    ///
+    /// For more detailed description of all the commands and return values,
+    /// refer to the stat man page [here](https://man7.org/linux/man-pages/man2/stat.2.html).
 
     pub fn fstat_syscall(&self, fd: i32, statbuf: &mut StatData) -> i32 {
+        // Attempt to get the file descriptor
```

Comment:
we've been adding 'BUG' as a comment here for the reason you mention in the panics section. Adding it here too could help with fixing that later on.

Reply:
got it, thanks for pointing this out